demo/api/src/products/product-importer.service.ts

```diff
@@ -0,0 +1,16 @@
+import { EntityManager } from "@mikro-orm/core";
+import { Injectable } from "@nestjs/common";
+import { LocalFileDataStream } from "@src/importer/data-streams/local-file-data-stream";
+
+import { ProductImporter } from "./product-importer";
+
+@Injectable()
+export class ProductImporterService {
+    constructor(private readonly em: EntityManager) {}
+
+    async createImporter(filePath: string) {
+        const importer = new ProductImporter(this.em);
```

Comment:
Since ProductImporter is injectable, it shouldn't be instantiated using `new`, but injected in the constructor instead.

Reply:
It should be the other way around.
It cannot be injectable since I need to create instances. This might change later

Reply:
Then we should remove the `@Injectable()` decorator from the importer for now.